src/components/robot/Client.ts

```diff
@@ -0,0 +1,234 @@
+  private client: RobotServiceClient;
+
+  constructor(init: RobotClientInit) {
+    if (isClient(init)) {
+      this.client = init.createServiceClient(RobotServiceClient);
+    } else if (isAddressInit(init)) {
+      this.client = new RobotServiceClient(init.serviceHost, init.opts);
+    } else {
+      throw new Error('invalid arguments');
+    }
+  }
+
+  private get robotService() {
+    return this.client;
+  }
+
+  // OPERATIONS
+
+  async getOperations() {
+    const robotService = this.robotService;
+    const request = new proto.GetOperationsRequest();
+    const response = await promisify<
+      proto.GetOperationsRequest,
+      proto.GetOperationsResponse
+    >(robotService.getOperations.bind(robotService), request);
+    return response.getOperationsList();
+  }
+
+  async cancelOperation(id: string) {
+    const robotService = this.robotService;
+    const request = new proto.CancelOperationRequest();
+    request.setId(id);
+    await promisify<
+      proto.CancelOperationRequest,
+      proto.CancelOperationResponse
+    >(robotService.cancelOperation.bind(robotService), request);
+  }
+
+  async blockForOperation(id: string) {
+    const robotService = this.robotService;
+    const request = new proto.BlockForOperationRequest();
+    request.setId(id);
+    await promisify<
+      proto.BlockForOperationRequest,
+      proto.BlockForOperationResponse
+    >(robotService.blockForOperation.bind(robotService), request);
+  }
+
+  async stopAll() {
+    const robotService = this.robotService;
+    const request = new proto.StopAllRequest();
+    await promisify<proto.StopAllRequest, proto.StopAllResponse>(
+      robotService.stopAll.bind(robotService),
+      request
+    );
+  }
+
+  // FRAME SYSTEM
+
+  async frameSystemConfig(transforms: Transform[]) {
+    const robotService = this.robotService;
+    const request = new proto.FrameSystemConfigRequest();
+    request.setSupplementalTransformsList(transforms);
+    const response = await promisify<
+      proto.FrameSystemConfigRequest,
+      proto.FrameSystemConfigResponse
+    >(robotService.frameSystemConfig.bind(robotService), request);
+    return response.getFrameSystemConfigsList();
+  }
+  async transformPose(
+    source: PoseInFrame,
+    destination: string,
+    supplementalTransforms: Transform[]
+  ) {
+    const robotService = this.robotService;
+    const request = new proto.TransformPoseRequest();
+    request.setSource(source);
+    request.setDestination(destination);
+    request.setSupplementalTransformsList(supplementalTransforms);
+    const response = await promisify<
+      proto.TransformPoseRequest,
+      proto.TransformPoseResponse
+    >(robotService.transformPose.bind(robotService), request);
+    const result = response.getPose();
+    if (!result) {
+      // eslint-disable-next-line no-warning-comments
+      // TODO: Can the response frame be undefined or null?
+      throw new Error('no pose');
+    }
```

Comment:
I'm not sure if getting a `null | undefined` pose is valid or should be an error.

Reply:
i asked a few people about this, and the loose consensus was that `null` pose should be an error